pyomo/contrib/doe/doe.py

```diff
@@ -1617,6 +1628,221 @@ def compute_FIM_full_factorial(
+            generate a regular nested for loop that can change from one to all the
+            design variables at a time. Default: True
+        file_name : str, optional
+            if provided, will save the results to a json file. Default: None
+
+        Returns
+        -------
+        factorial_results: dict
+            A dictionary containing the results of the factorial design with the
+            following keys:
+            - keys of model's experiment_inputs
+            - "log10(D-opt)": list of D-optimality values
+            - "log10(A-opt)": list of A-optimality values
+            - "log10(E-opt)": list of E-optimality values
+            - "log10(ME-opt)": list of ME-optimality values
+            - "solve_time": list of solve times
+            - "total_points": total number of points in the factorial design
+            - "success_count": number of successful runs
+            - "failure_count": number of failed runs
+            - "FIM_all": list of all FIMs computed for each point in the factorial
+              design.
+
+        Raises
+        ------
+        ValueError
+            If the design_values' keys are not a subset of the model's
+            `experiment_inputs` keys or if the design_values are not provided.
+        """
+
+        # Start timer
+        sp_timer = TicTocTimer()
+        sp_timer.tic(msg=None)
+        self.logger.info("Beginning Factorial Design.")
+
+        # Make new model for factorial design
+        self.factorial_model = self.experiment.get_labeled_model(
+            **self.get_labeled_model_args
+        ).clone()
+        model = self.factorial_model
+
+        if not design_values:
+            raise ValueError(
+                "design_values must be provided as a dictionary of lists "
+                "in the form {<'var_name'>: [<var_values>]}."
+            )
+
+        # Check whether the design_ranges keys are in the experiment_inputs
+        design_keys = set(design_values.keys())
+        map_keys = set([k.name for k in model.experiment_inputs.keys()])
+        if not design_keys.issubset(map_keys):
+            incorrect_given_keys = design_keys - map_keys
+            suggested_keys = map_keys - design_keys
+            raise ValueError(
+                f"design_values keys: {incorrect_given_keys} are incorrect."
+                f"The keys should be from the following keys: {suggested_keys}."
+            )
+
+        # Get the design map keys that match the design_values keys
+        design_map_keys = [
+            k for k in model.experiment_inputs.keys() if k.name in design_values.keys()
+        ]
+        # This ensures that the order of the design_values keys matches the order of the
+        # design_map_keys so that design_point can be constructed correctly in the loop.
+        des_ranges = [design_values[k.name] for k in design_map_keys]
+        if change_one_design_at_a_time:
+            factorial_points = generate_snake_zigzag_pattern(*des_ranges)
```

Comment:
Let's brainstorm different sensitivity analysis sequences we might use. We can then define an enum.